code/modules/surgery/organs/internal/cyberimp/augments_internal.dm

```diff
@@ -132,6 +137,15 @@
 		owner.SetKnockdown(0)
 		owner.SetImmobilized(0)
 		owner.SetParalyzed(0)
+		owner.setStaminaLoss(0)
+		ADD_TRAIT(owner, TRAIT_IGNOREDAMAGESLOWDOWN, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_IGNOREDAMAGESLOWDOWN, REF(src)), stun_resistance_time)
+		ADD_TRAIT(owner, TRAIT_BATON_RESISTANCE, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_BATON_RESISTANCE, REF(src)), stun_resistance_time)
+		ADD_TRAIT(owner, TRAIT_STUNIMMUNE, REF(src))
+		addtimer(TRAIT_CALLBACK_REMOVE(owner, TRAIT_STUNIMMUNE, REF(src)), stun_resistance_time)
+		organ_flags |= ORGAN_FAILING
+		addtimer(CALLBACK(src, PROC_REF(reboot)), implant_cooldown)
```

Comment:
Why does it make the organ fail when it triggers?

Reply:
From my understanding  it's the trigger for the implant, this one of the lines i didn't change.

/obj/item/organ/internal/cyberimp/brain/anti_stun/proc/on_signal(datum/source, amount)
	SIGNAL_HANDLER
	if(!(organ_flags & ORGAN_FAILING) && amount > 0)
		addtimer(CALLBACK(src, PROC_REF(clear_stuns)), stun_cap_amount, TIMER_UNIQUE|TIMER_OVERRIDE)

Reply:
It's kinda a misuse of the `ORGAN_FAILING`. Why not just track it with a cooldown?